src/server/src/test/java/io/cassandrareaper/resources/view/NodesStatusHttpTest.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2017-2017 Spotify AB
+ * Copyright 2017-2018 The Last Pickle Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cassandrareaper.resources.view;
+
+import io.cassandrareaper.management.http.HttpCassandraManagementProxy;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import com.datastax.mgmtapi.client.model.EndpointStates;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Covers {@link NodesStatus#NodesStatus(String, EndpointStates)}, which takes input data retrieved via the HTTP API.
+ *
+ * @see HttpCassandraManagementProxy#getNodesStatus()
+ */
+public class NodesStatusHttpTest {
+
+  @Test
+  public void testRegularValues() {
+    testSingleNode(
+        ImmutableMap.<String, String>builder()
+            .put("DC", "dc1")
+            .put("RACK", "rack1")
+            .put("ENDPOINT_IP", "10.0.0.1")
+            .put("IS_ALIVE", "true")
+            .put("STATUS_WITH_PORT", "NORMAL,-1181245519517832414")
```

Comment:
I copied the output of an actual Cassandra instance. I'm not sure why it's formatted that way, or if it even refers to "port" in the network sense. At the end of the day it doesn't matter much since we discard everything after the comma.

Reply:
Weird, OK. I'll resolve this as it seems that you've done things the right way.